### PR TITLE
Add notification service

### PR DESCRIPTION
**Pull request by jose-gataca**

### CODE REVIEW

File: models/notification.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+)
+
+type GatcNotification struct {
+	Type     NotificationType `json:"-,omitempty"`
+	Source   string           `json:"-,omitempty"`
+	Receiver string           `json:"-,omitempty"`
+	Data     interface{}      `json:"data"`
+}
+
+type NotMessageData struct {
```
Comment (samu-gataca) [inline models/notification.go:24]: What is this?

File: models/notification.go [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+	return n == MESSAGE || n == CREDENTIAL_VALIDATED || n == CREDENTIAL_SUSPENDED || n == CREDENTIAL_REVOKED || n == SERVICE_REVOKED
+}
+
+const (
+	MESSAGE              NotificationType = "message"
+	CREDENTIAL_VALIDATED NotificationType = "credential_validated"
+	CREDENTIAL_SUSPENDED NotificationType = "credential_suspended"
+	CREDENTIAL_REVOKED   NotificationType = "credential_revoked"
+	SERVICE_REVOKED      NotificationType = "service_revoked"
+)
```
Comment (samu-gataca) [inline models/notification.go:15]: Enumerated for each type of message?

File: service/notificationService.go [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+	"encoding/json"
+	"net/http"
+
+	"github.com/gataca-io/gataca-core/log"
+	"github.com/gataca-io/gataca-core/models"
+	"github.com/labstack/echo"
+)
+
+const (
+	ROUTE_NOTIFICATIONS = "/api/v1/notifications"
```
Comment (samu-gataca) [inline service/notificationService.go:14]: Protected? I suppose it needs to be protected. Any issuer could use it ... Ok as 1st version

**Review by samu-gataca — APPROVED** [review]

File: models/notification.go (hunk 1 above)
Comment (jose-gataca) [reply models/notification.go:24]: This was on the backbone, it has been moved. The not stands as an abreviation of "notification".
It's the types for our notifications interface

File: models/notification.go (hunk 2 above)
Comment (jose-gataca) [reply models/notification.go:15]: For each type of notification, yes.

File: service/notificationService.go (hunk 3 above)
Comment (jose-gataca) [reply service/notificationService.go:14]: Any issuer can/shall use it know. With the changes of this sprint, the issuer requests a notification when the credential is ready.
The polling is working no more.

**Pull request merged**